webapp/channels/src/components/color_input.tsx

```diff
@@ -13,172 +13,155 @@ type Props = {
     isDisabled?: boolean;
 }
 
-type State = {
-    focused: boolean;
-    isOpened: boolean;
-    value: string;
-}
-
-export default class ColorInput extends React.PureComponent<Props, State> {
-    private colorPicker: React.RefObject<HTMLDivElement>;
-    private colorInput: React.RefObject<HTMLInputElement>;
-
-    public constructor(props: Props) {
-        super(props);
-        this.colorPicker = React.createRef();
-        this.colorInput = React.createRef();
-
-        this.state = {
-            focused: false,
-            isOpened: false,
-            value: props.value,
-        };
-    }
-
-    static getDerivedStateFromProps(props: Props, state: State) {
-        if (!state.focused && props.value !== state.value) {
-            return {
-                value: props.value,
-            };
-        }
-
-        return null;
+const ColorInput = ({
+    id,
+    onChange: onChangeFromProps,
+    value: valueFromProps,
+    isDisabled,
+}: Props) => {
+    const colorPicker = useRef<HTMLDivElement>(null);
+    const colorInput = useRef<HTMLInputElement>(null);
+
+    const [isFocused, setIsFocused] = useState(false);
+    const [isOpened, setIsOpened] = useState(false);
+    const [valueFromState, setValueFromState] = useState(valueFromProps);
+
+    if (!isFocused && valueFromProps !== valueFromState) {
+        setValueFromState(valueFromProps);
```

Comment:
Do we still need this when we do something similar in the `onBlur` handler? I'm not sure if this was added for some case like when you close the picker or something, but it seems odd to be doing this in this way. I don't know if setting state from within the body of a function component would cause an actual problem or if it just looks unusual